ipa-core/src/protocol/ipa_prf/malicious_security/verifier.rs

```diff
@@ -8,38 +8,35 @@ use crate::{
 };
 
 #[allow(non_upper_case_globals)]
-pub struct ProofVerifier<F: PrimeField> {
-    zero_share: F,
-}
+pub struct ProofVerifier {}
 
 ///
 /// Distributed Zero Knowledge Proofs algorithm drawn from
 /// `https://eprint.iacr.org/2023/909.pdf`
 ///
 #[allow(non_upper_case_globals)]
-impl<F> ProofVerifier<F>
-where
-    F: PrimeField,
-{
-    /// This function outputs an `out_share` and a `ProofVerifier` that consists of a `zero_share`.
-    pub fn verify_proof<const λ: usize, const P: usize>(
-        out_share: F,
+impl ProofVerifier {
+    /// This function interprets the zkp as points on a polynomial, and interpolates the
+    /// value of this polynomial at the provided value of `r`
+    pub fn interpolate_at_r<F: PrimeField, const λ: usize, const P: usize>(
         zkp: &[F; P],
         r: F,
-    ) -> (F, Self) {
-        let denominator_g = CanonicalLagrangeDenominator::<F, P>::new();
-        let lagrange_table_g = LagrangeTable::<F, P, 1>::new(&denominator_g, &r);
-        let g_r_share = lagrange_table_g.eval(zkp)[0];
-        let sum_share = (0..λ).fold(F::ZERO, |acc, i| acc + zkp[i]);
-
-        // Reveal `b_share` to one another to reconstruct `b` and check if `b = 0`. If the check doesn't pass, abort.
-        let zero_share = sum_share - out_share;
+        lagrange_denominator: &CanonicalLagrangeDenominator<F, P>,
+    ) -> F {
+        let lagrange_table_g = LagrangeTable::<F, P, 1>::new(lagrange_denominator, &r);
+        lagrange_table_g.eval(zkp)[0]
+    }
 
-        (g_r_share, Self { zero_share })
+    /// This function outputs an `out_share` and a `ProofVerifier` that consists of a `zero_share`.
```

Comment:
this seems to produce only a single field value

Reply:
Comment is wrong... will fix.